execution_engine/src/execution/executor.rs

```diff
@@ -203,12 +206,14 @@ impl Executor {
                 effects: runtime.context().effects(),
                 transfers: runtime.context().transfers().to_owned(),
                 cost: runtime.context().gas_counter(),
+                messages: runtime.context().messages(),
             },
             Err(error) => ExecutionResult::Failure {
                 effects,
                 error: error.into(),
                 transfers: runtime.context().transfers().to_owned(),
                 cost: runtime.context().gas_counter(),
+                messages,
```

Comment:
is there any particular reason this uses `messages` from `TrackingCopy` rather than `Runtime`?

Reply:
In this case not really since the standard payment logic wouldn't emit any messages, but in the future it might. Every time a message is emitted, there is an impact on global state in the form of a write transform to `Key::message(entity_hash, topic_name_hash, message_index)`. In this case if there is a failure for the standard payment, the effects of that call are not included in the execution result (it returns the effects before the call to `call_host_standard_payment`) so it's natural to not include any messages that were produced by that call since the hashes for the messages won't be added to GS and in turn the messages would not be verifiable.